apigo/api/mappings.go

```diff
@@ -0,0 +1,325 @@
+
+	resp, err := http.DefaultClient.Do(graphqlQuery)
+	if err != nil {
+		return nil, err
+	}
+	switch resp.StatusCode {
+	case http.StatusOK:
+		// If everything went well, we go on
+	case http.StatusUnauthorized:
+		return nil, &invalidTokenError{message: "Token is invalid", statusCode: http.StatusUnauthorized}
+	case http.StatusForbidden:
+		return nil, &invalidTokenError{message: "You don't have rights to perform this action", statusCode: http.StatusForbidden}
+	default:
+		// Return other errors
+		return nil, errors.New("Error while requesting mapping from pyrog")
+	}
+	defer resp.Body.Close()
+
+	gqlResp := graphqlResponse{}
+	err = json.NewDecoder(resp.Body).Decode(&gqlResp)
+	if err != nil {
+		return nil, err
+	}
+
+	resourceMapping := gqlResp.Data.Resource
+
+	// Dereference concept maps
+	err = dereferenceConceptMap(&resourceMapping, authorizationHeader)
+	if err != nil {
+		return nil, err
+	}
+
+	return &resourceMapping, nil
+}
+
+func dereferenceConceptMap(mapping *mappingResource, authorizationHeader string) error {
+	for _, attribute := range mapping.Attributes {
+		for _, inputGroup := range attribute.InputGroups {
+			for _, input := range inputGroup.Inputs {
+				if input.ConceptMapID != "" {
+					conceptMap, err := fetchConceptMap(input.ConceptMapID, authorizationHeader)
+					if err != nil {
+						return err
+					}
+					input.ConceptMap = conceptMap
+				}
+			}
+		}
+	}
+	return nil
+}
+
+type conceptMapResponse struct {
+	Group []struct {
+		Element []struct {
+			Code   string `json:"code"`
+			Target []struct {
+				Equivalence string `json:"equivalence"`
+				Code        string `json:"code"`
+			} `json:"target"`
+		} `json:"element"`
+	} `json:"group"`
+}
+
+func fetchConceptMap(conceptMapID string, authorizationHeader string) (map[string]string, error) {
+	url := fmt.Sprintf("%s/ConceptMap/%s", fhirURL, conceptMapID)
+	req, _ := http.NewRequest("GET", url, nil)
+	req.Header.Set("Authorization", authorizationHeader)
+	req.Header.Set("Content-Type", "application/json")
+
+	resp, err := http.DefaultClient.Do(req)
+	if err != nil {
+		return nil, err
+	}
+	switch resp.StatusCode {
+	case http.StatusOK:
+		// If everything went well, we go on
+	case http.StatusUnauthorized:
+		return nil, &invalidTokenError{message: "Token is invalid", statusCode: http.StatusUnauthorized}
+	case http.StatusForbidden:
+		return nil, &invalidTokenError{message: "You don't have rights to perform this action", statusCode: http.StatusForbidden}
+	default:
+		// Return other errors
+		return nil, errors.New("Error while fetching concept map")
+	}
+
+	conceptMapResp := conceptMapResponse{}
+	err = json.NewDecoder(resp.Body).Decode(&conceptMapResp)
+	if err != nil {
+		return nil, err
+	}
+
+	// Turn the concept map into a map
+	conceptMap := make(map[string]string)
+	for _, group := range conceptMapResp.Group {
+		for _, element := range group.Element {
+			// NOTE we only handle a single target for each source
```

Comment:
does this deserve an issue ?

Reply:
Done here: https://github.com/arkhn/fhir-river/issues/134